### DISCUSSION TITLE
Help understanding permissions

### DISCUSSION THREAD

**Discussion by realSConway** [containers/podman#17540] [General]
Hi folks,

I need help/assistance on why a file is not accessible to another rootless container running in same pod.

The two projects im using are: 
- osminogin/docker-tor-simple
- kylemanna/bitcoind

Warning in question is found in btc debug.log:
`Authentication cookie /var/lib/tor/control_auth_cookie could not be opened (check permissions)`

Containers are run as:
```
podman run --rm --name tor --pod bitcoin --volume ./tor:/var/lib/tor:z --volume ./tor/torrc:/etc/tor/torrc:ro osminogin/tor-simple
```
```
podman run --rm --name btc --pod bitcoin --volume ./bitcoin:/bitcoin/.bitcoin:z kylemanna/bitcoind
```
tor has following user/group:
```
podman exec -ti tor id
uid=100(tor) gid=65533(nogroup) groups=65533(nogroup)
```
and btc:
 ```
podman exec -ti btc id
uid=0(root) gid=0(root) groups=0(root)
```

permissions to `./tor`
```
ls -l ./tor
total 11544
-rw-------. 1 100099 165532      51 Feb 16 20:38 .ash_history
-rw-------. 1 100099 165532   18209 Feb 16 20:20 cached-certs
-rw-------. 1 100099 165532 2218992 Feb 16 20:20 cached-microdesc-consensus
-rw-------. 1 100099 165532 9533971 Feb 16 20:23 cached-microdescs
-rw-------. 1 100099 165532    7573 Feb 16 20:23 cached-microdescs.new
-rw-r-----. 1 100099 165532      32 Feb 16 20:54 control_auth_cookie
drwx------. 1 100099 165532       0 Feb 16 20:20 keys
-rw-------. 1 100099 165532       0 Feb 16 20:54 lock
-rw-------. 1 100099 165532   13977 Feb 16 21:23 state
-rw-r--r--. 1 100099 165532    9035 Feb 16 17:49 torrc
```
I've tried mounting tor volume (`./tor:/var/lib/tor`) with `:z`, `:Z` `:z,U` `--group-add keep-groups` or with `-e PUID=100` but error is still logged.

`podman unshare chown 100: ./tor` did nothing either.

Shouldn't a uid 0 be able to read/write to a uid 100?

Any pointers would be greatly appreciated

**Comment by Luap99** [comment]
maybe https://www.redhat.com/sysadmin/debug-rootless-podman-mounted-volumes and https://www.redhat.com/sysadmin/container-permission-denied-errors will help
